### PR TITLE
Date Range Table Filter (New Feature)

### PR DESCRIPTION
**Pull request by ghost**
Concept for a new feature that been discussed/requested in a few issue trackers. Well need testing.

### CODE REVIEW

File: src/plugins/tables/tables.js
Comment (RobJohnston) [inline src/plugins/tables/tables.js:251]: Maxium -> Maximum

File: src/plugins/tables/tables.js
Comment (RobJohnston) [inline src/plugins/tables/tables.js:243]: Minium -> Minimum

**Comment by thekodester** [comment]
FYI: I only have done basic testing with Safari.

I would recommend testing:
- [ ] Different date formats in the table (most likely to cause issues)
- [ ] Combined with additional filters (selects, checkboxes)
- [ ] Additional web browsers (Firefox, IE, Chrome)

**Review by GormFrank — APPROVED** [review]
Despite all the possible date format that could cause issue, this is acceptable. It passed all my tests and therefore I approve! Oh and for what it's worth, IE11 isn't that much of a concern anymore since it is under our threshold of 5% (see [WET-BOEW's DD-2](https://wet-boew.github.io/wet-boew-documentation/decision/2.html) for more info).

**Comment by GormFrank** [comment]
@thekodester  What have you done in your most recent force push?

**Comment by thekodester** [comment]
@GormFrank No changes, I run an automated script that updates all my branches to latest commits of WET-BOEW. When I do any testing with my fork (thekodester/wet-boew).

File: src/plugins/tables/tables.js
Comment (duboisp) [inline src/plugins/tables/tables.js:259]: We would need to documents the date format the author need to put into the data table in order to take advantage of this date filtering. Like:
* 2016-07-26 14:07:00
* 2016-07-26T14:07:00
* 2016-07-26

That documentation can be added in here : https://github.com/wet-boew/wet-boew/tree/master/site/pages/docs/ref/tables  or into the working example

**Review by duboisp — APPROVED** [review]

**Comment by thekodester** [comment]
There is a debate of which PR we submit first. As #8965 changes how the filter examples are displayed and documented. That if you even go forward with #8965, but if we do I would debate it make sense to do #8965 first and update this PR accordingly.

**Comment by GormFrank** [comment]
@thekodester I could see this one being merged first as it follows a standard date format that we agreed on & approved and it is technically ready to be merged. The issue is that the documentation updates are in #8965. It really depends on what is your timeline to get this Date range table filter added.

**Comment by thekodester** [comment]
I’m not on any deadline, I do this voluntarily.

But I know there been interest in this feature in the past from others like @michael-milette

The numbers filter currently has the most recent interest

**Comment by GormFrank** [comment]
I approved #8965 so we'll be waiting for @duboisp stamp of approval too and it will be merged first. Thanks @thekodester !

**Comment by thekodester** [comment]
@GormFrank Sounds good, this PR can be updated accordingly afterwards.

**Comment by GormFrank** [comment]
@thekodester The other PR has been merged :)

**Comment by thekodester** [comment]
Might wanna do a quick test but I merged the latest updates and added the documentation.

**Comment by thekodester** [comment]
Retested filtered by date, drop-down, and/or checkboxes with no conflicts.

**Comment by thekodester** [comment]
New Documentation is:

For date range create two date inputs (`<input type="date"/>`) add the following attribute `data-column="COLUMN_NUMBER"`
Example: `<input type="date" class="form-control" id="dt_mindate" name="dt_mindate" data-column="1"/>`
Each date input requires the same column number. Please note that the column numbers start at 0.

---

Accepted Formats for Date Range
Currently only accepts ISO 8601 (YYYY-MM-DD). See examples below.
2016-07-26T14:07:00
2016-07-26 14:07:00
2016-07-26

Screenshot:
Screen Shot 2020-12-11 at 4 42 05 PM

**Review by GormFrank — APPROVED** [review]
That's awesome! Thank you @thekodester

**Pull request merged**
